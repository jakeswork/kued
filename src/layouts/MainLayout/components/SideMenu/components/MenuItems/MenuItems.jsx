import React, { Fragment } from 'react';

import Button from '../../../../../../components/Button';

export default () => (
  <Fragment>
    <Button flat style={{ color: 'white' }} link="/feed">
      Warmane
    </Button>
    <Button flat style={{ color: 'white' }} link="/vote">
      Other Servers
    </Button>
    <Button flat style={{ color: 'white' }} link="/contact">
      Contact
    </Button>
  </Fragment>
);
