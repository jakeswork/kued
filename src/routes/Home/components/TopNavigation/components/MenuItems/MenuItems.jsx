import React, { Fragment } from 'react';

import Button from '../../../../../../components/Button';

export default () => (
  <Fragment>
    <Button flat style={{ color: 'white' }} link="/feed">
      Whos Playing
    </Button>
    <Button flat style={{ color: 'white' }} link="/vote">
      Compatibility
    </Button>
    <Button flat style={{ color: 'white' }} link="/contact">
      Contact
    </Button>
  </Fragment>
);
