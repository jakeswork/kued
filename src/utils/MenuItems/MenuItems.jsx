import React, { Fragment } from 'react';
import { node, string } from 'prop-types';

import Button from '../../components/Button';
import propTypes from './types';

const MenuItems = ({ classes }) => {
  const NavButton = ({ children, link }) => (
    <Button
      flat
      activeClassName={classes.active}
      link={link}
    >
      { children }
    </Button>
  );

  NavButton.propTypes = {
    children: node.isRequired,
    link: string.isRequired,
  };

  return (
    <Fragment>
      <NavButton link="/playing">
        Live Feed
      </NavButton>
      <NavButton link="/servers">
        Servers
      </NavButton>
      <NavButton link="/support">
        Support
      </NavButton>
    </Fragment>
  );
};

MenuItems.propTypes = propTypes;

export default MenuItems;
