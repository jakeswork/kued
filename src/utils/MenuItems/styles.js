export default theme => ({
  active: {
    color: theme.colorPrimary,
    cursor: 'default',
    pointerEvents: 'none',
  },
});
