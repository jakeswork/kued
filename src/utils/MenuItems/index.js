import withStyles from 'react-jss';

import MenuItems from './MenuItems';
import styles from './styles';

export default withStyles(styles)(MenuItems);
