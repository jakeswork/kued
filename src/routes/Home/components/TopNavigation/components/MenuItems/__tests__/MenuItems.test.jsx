import React from 'react';
import { shallow } from 'enzyme';

import MenuItems from '../MenuItems';

describe('The MenuItems instance', () => {
  describe('when rendered', () => {
    it('should return the list of buttons', () => {
      const wrapper = shallow(<MenuItems />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
