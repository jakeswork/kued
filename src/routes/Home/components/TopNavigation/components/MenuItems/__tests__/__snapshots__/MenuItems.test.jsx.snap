// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The MenuItems instance when rendered should return the list of buttons 1`] = `
<Fragment>
  <Jss(Button)
    flat={true}
    icon={null}
    link="/feed"
    secondary={false}
    style={
      Object {
        "color": "white",
      }
    }
  >
    Whos Playing
  </Jss(Button)>
  <Jss(Button)
    flat={true}
    icon={null}
    link="/vote"
    secondary={false}
    style={
      Object {
        "color": "white",
      }
    }
  >
    Compatibility
  </Jss(Button)>
  <Jss(Button)
    flat={true}
    icon={null}
    link="/contact"
    secondary={false}
    style={
      Object {
        "color": "white",
      }
    }
  >
    Contact
  </Jss(Button)>
</Fragment>
`;
